### PR TITLE
r/glue_workflow - add tagging support + arn + max concurrent runs

### LINKED ISSUE #14994: Support for Glue Workflow Max Concurrency
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
`August 12, 2020` [Glue Added Support for max concurrency workflow runs](https://aws.amazon.com/about-aws/whats-new/2020/08/aws-glue-provides-ability-stop-restart-glue-workflows/). My team spins up all of our Data Lake workloads using Terraform. We want to specify the new parameter as it will add predictability to our workflows.

[This is supported in Boto3 now](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html#Glue.Client.create_workflow)
```python
response = client.create_workflow(
    Name='string',
    Description='string',
    DefaultRunProperties={
        'string': 'string'
    },
    Tags={
        'string': 'string'
    },
    MaxConcurrentRuns=123
)
```

### New or Affected Resource(s)

* aws_glue_workflow

### Potential Terraform Configuration

```hcl
resource "aws_glue_workflow" "example" {
  name = "example"
  description = ""
  max_concurrent_runs = 1 // Default is unlimited 
}
```

### References

* https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html#Glue.Client.create_workflow
* https://aws.amazon.com/about-aws/whats-new/2020/08/aws-glue-provides-ability-stop-restart-glue-workflows/
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/glue_workflow

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #13826
Closes #14994

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_glue_workflow - add `tags` argument
resource_aws_glue_workflow - add `arn` attribute
resource_aws_glue_workflow - add `max_concurrent_runs` argument
resource_aws_glue_workflow - plan time validation for `name`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueWorkflow_'
--- PASS: TestAccAWSGlueWorkflow_DefaultRunProperties (44.11s)
--- PASS: TestAccAWSGlueWorkflow_Description (80.65s)
--- PASS: TestAccAWSGlueWorkflow_Tags (103.40s)
--- PASS: TestAccAWSGlueWorkflow_disappears (30.25s)
--- PASS: TestAccAWSGlueWorkflow_maxConcurrentRuns (107.44s)
```

### CODE REVIEW

File: aws/resource_aws_glue_workflow.go [hunk 1]
```diff
@@ -71,20 +84,21 @@ func resourceAwsGlueWorkflowCreate(d *schema.ResourceData, meta interface{}) err
...
 func resourceAwsGlueWorkflowRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).glueconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
 	input := &glue.GetWorkflowInput{
 		Name: aws.String(d.Id()),
 	}
 
-	log.Printf("[DEBUG] Reading Glue Workflow: %s", input)
+	log.Printf("[DEBUG] Reading Glue Workflow: %#v", input)
```
Comment (ewbankkit) [inline aws/resource_aws_glue_workflow.go:93]: I think we prefer `%s` as the AWS SDK generates `.String()` methods.

Comment (DrFaust92) [reply aws/resource_aws_glue_workflow.go:93]: I've used this notation previously (as i've seen in being used somewhere in the provider) and this was previously approved. so idk? i dont really mind which.

Comment (ewbankkit) [reply aws/resource_aws_glue_workflow.go:93]: No need to change then.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSGlueWorkflow_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueWorkflow_ -timeout 120m
=== RUN   TestAccAWSGlueWorkflow_basic
=== PAUSE TestAccAWSGlueWorkflow_basic
=== RUN   TestAccAWSGlueWorkflow_maxConcurrentRuns
=== PAUSE TestAccAWSGlueWorkflow_maxConcurrentRuns
=== RUN   TestAccAWSGlueWorkflow_DefaultRunProperties
=== PAUSE TestAccAWSGlueWorkflow_DefaultRunProperties
=== RUN   TestAccAWSGlueWorkflow_Description
=== PAUSE TestAccAWSGlueWorkflow_Description
=== RUN   TestAccAWSGlueWorkflow_Tags
=== PAUSE TestAccAWSGlueWorkflow_Tags
=== RUN   TestAccAWSGlueWorkflow_disappears
=== PAUSE TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_basic
=== CONT  TestAccAWSGlueWorkflow_Description
=== CONT  TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_DefaultRunProperties
=== CONT  TestAccAWSGlueWorkflow_Tags
=== CONT  TestAccAWSGlueWorkflow_maxConcurrentRuns
=== CONT  TestAccAWSGlueWorkflow_disappears
    resource_aws_glue_workflow_test.go:230: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueWorkflow_disappears (22.96s)
--- PASS: TestAccAWSGlueWorkflow_basic (32.73s)
--- PASS: TestAccAWSGlueWorkflow_DefaultRunProperties (33.19s)
--- PASS: TestAccAWSGlueWorkflow_Description (47.56s)
--- PASS: TestAccAWSGlueWorkflow_Tags (59.36s)
--- PASS: TestAccAWSGlueWorkflow_maxConcurrentRuns (60.90s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	60.945s
```

**Comment by breathingdust** [comment]
LGTM! 🚀  Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TESTARGS='-run=TestAccAWSGlueWorkflow_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueWorkflow_ -timeout 120m
=== RUN   TestAccAWSGlueWorkflow_basic
=== PAUSE TestAccAWSGlueWorkflow_basic
=== RUN   TestAccAWSGlueWorkflow_maxConcurrentRuns
=== PAUSE TestAccAWSGlueWorkflow_maxConcurrentRuns
=== RUN   TestAccAWSGlueWorkflow_DefaultRunProperties
=== PAUSE TestAccAWSGlueWorkflow_DefaultRunProperties
=== RUN   TestAccAWSGlueWorkflow_Description
=== PAUSE TestAccAWSGlueWorkflow_Description
=== RUN   TestAccAWSGlueWorkflow_Tags
=== PAUSE TestAccAWSGlueWorkflow_Tags
=== RUN   TestAccAWSGlueWorkflow_disappears
=== PAUSE TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_basic
=== CONT  TestAccAWSGlueWorkflow_Tags
=== CONT  TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_DefaultRunProperties
=== CONT  TestAccAWSGlueWorkflow_Description
=== CONT  TestAccAWSGlueWorkflow_maxConcurrentRuns
=== CONT  TestAccAWSGlueWorkflow_disappears
    resource_aws_glue_workflow_test.go:230: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueWorkflow_disappears (13.11s)
--- PASS: TestAccAWSGlueWorkflow_basic (17.40s)
--- PASS: TestAccAWSGlueWorkflow_DefaultRunProperties (17.43s)
--- PASS: TestAccAWSGlueWorkflow_Description (27.43s)
--- PASS: TestAccAWSGlueWorkflow_Tags (43.07s)
--- PASS: TestAccAWSGlueWorkflow_maxConcurrentRuns (51.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	52.947s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TESTARGS='-run=TestAccAWSGlueWorkflow_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueWorkflow_ -timeout 120m
=== RUN   TestAccAWSGlueWorkflow_basic
=== PAUSE TestAccAWSGlueWorkflow_basic
=== RUN   TestAccAWSGlueWorkflow_maxConcurrentRuns
=== PAUSE TestAccAWSGlueWorkflow_maxConcurrentRuns
=== RUN   TestAccAWSGlueWorkflow_DefaultRunProperties
=== PAUSE TestAccAWSGlueWorkflow_DefaultRunProperties
=== RUN   TestAccAWSGlueWorkflow_Description
=== PAUSE TestAccAWSGlueWorkflow_Description
=== RUN   TestAccAWSGlueWorkflow_Tags
=== PAUSE TestAccAWSGlueWorkflow_Tags
=== RUN   TestAccAWSGlueWorkflow_disappears
=== PAUSE TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_basic
=== CONT  TestAccAWSGlueWorkflow_Tags
=== CONT  TestAccAWSGlueWorkflow_disappears
=== CONT  TestAccAWSGlueWorkflow_DefaultRunProperties
=== CONT  TestAccAWSGlueWorkflow_Description
=== CONT  TestAccAWSGlueWorkflow_maxConcurrentRuns
=== CONT  TestAccAWSGlueWorkflow_disappears
    resource_aws_glue_workflow_test.go:230: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSGlueWorkflow_disappears (11.91s)
--- PASS: TestAccAWSGlueWorkflow_basic (14.65s)
--- PASS: TestAccAWSGlueWorkflow_DefaultRunProperties (17.82s)
--- PASS: TestAccAWSGlueWorkflow_Description (27.70s)
--- PASS: TestAccAWSGlueWorkflow_Tags (37.83s)
--- PASS: TestAccAWSGlueWorkflow_maxConcurrentRuns (47.60s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	48.918s
```

**Pull request merged**